autopush/websocket.py

```diff
@@ -45,6 +45,33 @@ class SimplePushServerProtocol(WebSocketServerProtocol):
     # Testing purposes
     parent_class = WebSocketServerProtocol
 
+    # Defer helpers
+    def deferToThread(self, func, *args, **kwargs):
+        d = deferToThread(func, *args, **kwargs)
+        self._callbacks.append(d)
+
+        def f(result):
+            if d in self._callbacks:
+                self._callbacks.remove(d)
+            return result
+        d.addBoth(f)
```

Comment:
Ah, [I see](https://twistedmatrix.com/documents/current/core/howto/defer.html#visual-explanation) returning failures is equivalent to throwing. */me notes for the future*